### PR TITLE
Add a google sheets connection

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request

* This creates a new connection to Google Sheets from /marketing/connections
* Needs D35254
* This has no useful function yet

#### Testing instructions

* Checkout this branch
* Apply the patch D35254
* Go to http://calypso.localhost:3000/marketing/connections
* Select a site
* Scroll down to Google Sheets and click connect
* You should see a success message and your Google avatar below the connection
* Click disconnect
* You should see another success message

Screenshots:
Screenshot 2019-10-22 at 13 44 19
Screenshot 2019-10-22 at 13 44 08

### CODE REVIEW

**Review by davemart-in — APPROVED** [review]
Looks good to me visually. I made some updates in https://github.com/Automattic/wp-calypso/commit/4f33745080316dfec2e19940c7fd5486a2af77ca to add the three images and to add both the form and the sheets screenshot.

**Preview**
Screen Shot 2019-10-22 at 12 02 14 PM-1

File: client/my-sites/marketing/connections/services/google-drive.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	/**
+	 * Deletes the passed connections.
+	 *
+	 * @param {Array} connections Optional. Connections to be deleted.
+	 *                            Default: All connections for this service.
+	 */
+	removeConnection = () => {
+		this.setState( { isDisconnecting: true } );
+		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
```
Comment (creativecoder) [inline client/my-sites/marketing/connections/services/google-drive.js]: See how `google_my_business` site keyrings are disconnected when a user triggers the "Disconnect" action:  https://github.com/Automattic/wp-calypso/blob/add/google-sheets-connection/client/my-sites/marketing/connections/services/google-my-business.js#L71

**Review by creativecoder — CHANGES_REQUESTED** [review]
Now that we've decided on using a site keyring for Google Drive, let's make sure we create and delete site keyrings when a connection is made. See comments for details.

**Comment by scruffian** [comment]
Putting this on hold until we have a clearer idea of how to proceed

**Pull request closed without merging**